### DISCUSSION TITLE
Is it possible to enable breadcrumbs during consult preview?

### DISCUSSION THREAD

**Discussion by amosbird** [minad/consult#1056] [Q&A] [answered]
This feature would be useful for navigating large codebases by showing which method a previewed line belongs to. Currently, the preview buffer does not display breadcrumbs. Should I add a hook to enable it?

**Comment by minad** [comment, marked as answer]
I believe consult-preview-allowed-hooks should be able to handle this. See the examples at https://github.com/minad/consult?tab=readme-ov-file#live-previews.

~~~ elisp
(add-to-list 'consult-preview-allowed-hooks 'breadcrumb-local-mode)
~~~

**Reply by amosbird** [reply]
I tried the following:

```elisp
(add-to-list 'consult-preview-allowed-hooks 'prog-mode)
(add-to-list 'consult-preview-allowed-hooks 'c++-ts-mode)
(add-to-list 'consult-preview-allowed-hooks 'breadcrumb-local-mode)
```

However, it still doesn't work.

**Reply by minad** [reply]
Well, trying blindly doesn't usually lead to a working result (what are prog-mode and c++-ts-mode doing there?). Did you take a look at the documentation and the examples? `breadcrumb-local-mode` seems to be the one you want to allow. But note that you also have to add `breadcrumb-local-mode` to the respective major mode hook in the first place to enable it.

**Reply by amosbird** [reply]
> Well, trying blindly doesn't usually lead to a working result

Yes.

After examining how `consult-preview` works, it turns out I need to do the following:

`(setq consult-preview-variables (assq-delete-all 'delay-mode-hooks consult-preview-variables))`

**Reply by minad** [reply]
No, this is not the right thing. If you disable `delay-mode-hooks`, all hooks will be enabled during preview, which is usually not desired. Consult provides `consult-preview-allowed-hooks` specifically in order to run certain hooks despite `delay-mode-hooks=t`:

https://github.com/minad/consult/blob/41d382c9da9a84ab5391ee906e24f7126d03bd53/consult.el#L1341-L1346

**Reply by amosbird** [reply]
Oops, I should have realized earlier that Doom Emacs doesn't have a recent enough version of the Consult package. I don't have the changes from [this commit](https://github.com/minad/consult/commit/369592fe2f160c1b54f91266104992fc25bc5e14), which means `consult-preview-allowed-hooks` only handles `find-file-hook`.

Thanks for helping me out!
